jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Callback.java

```diff
@@ -303,12 +328,12 @@ public void succeeded()
                 try
                 {
                     completed.run();
-                    callback.succeeded();
                 }
                 catch (Throwable t)
                 {
-                    callback.failed(t);
+                    Callback.failed(callback, t);
                 }
+                callback.succeeded();
```

Comment:
This should be moved back into the try, otherwise we fail the callback and then succeed it.

Okay may be not, but it's confusing to have it here, as my comment above says.